edx_name_affirmation/views.py

```diff
@@ -125,6 +126,35 @@ def post(self, request):
         return Response(status=response_status, data=data)
 
 
+class VerifiedNameHistoryView(AuthenticatedAPIView):
+    """
+    Endpoint for VerifiedName history.
+    /edx_name_affirmation/v1/verified_name/history?username=xxx
+
+    Supports:
+        HTTP GET: Return a list of VerifiedNames for the given user.
+    """
+    def get(self, request):
+        """
+        Get a list of verified name objects for the given user, ordered by most recently created.
+        """
+        username = request.GET.get('username')
+        if username and not request.user.is_staff:
+            return Response(
+                status=status.HTTP_403_FORBIDDEN,
+                data={"detail": "Must be a Staff User to Perform this request."}
+            )
+
+        user = get_user_model().objects.get(username=username) if username else request.user
+        verified_name_qs = get_verified_name_history(user)
+
+        serialized_data = [
+            VerifiedNameSerializer(verified_name).data for verified_name in verified_name_qs
+        ]
+
+        return Response(serialized_data)
```

Comment:
It may be more flexible to return an dictionary instead of a list. If the Response contains just a list, adding additional metadata to the response later (e.g. if we feel the need to paginate it) would be a breaking change. Returning something like `{ results: <list> }` is more flexible because it avoids this issue.